src/virtualenvrunner/pythonversionrun.py

```diff
@@ -20,16 +20,22 @@ def _get_run_lambda(self, python_version):
         return lambda: self._run_with_python_version(python_version)
 
     def _run_with_python_version(self, python_version):
-        self._run(pythonexe='python{major}.{minor}{exe_suffix}'.format(
-            major=python_version.major,
-            minor=python_version.minor,
-            exe_suffix=get_exe_suffix()),
-                  python_version=python_version)
+        if python_version.minor:
```

Comment:
Instead of `if .. else` it would be better to use a helper function, same way like `get_exe_suffix`.

Even better solution would be to treat version as one entity and implement its string representation in PythonVersion class, something like below:

```
class PythonVersion:
    def __init__(self, major, minor):
        self.major = major
        self.minor = minor

    def __str__(self):
        if not self.minor:
            return str(self.major)
        return '{}.{}'.format(self.major, self.minor)

def get_python_versions():
    for v in [(2, None), (2, 7), (3, None), (3, 4), (3, 5), (3, 6)]:
        yield PythonVersion(*v)

for v in get_python_versions():
    print('python{version}'.format(version=v))
```

Reply:
Refactored